tests/ftests/systemd.py

```diff
@@ -27,3 +29,91 @@ def is_delegated(config, scope_name):
                 # until we figure out something better :(
                 return True
             raise re
+
+    # This function builds the absolute path of the cgroup, using the default
+    # systemd path and checks if the cgroup exists. The function can build
+    # path for all three cgroup setup modes, including empty cgroup v2.
+    @staticmethod
+    def cgroup_exists(config, ctrl_name, cgroup_name, ignore_systemd=False):
+        # 'None' is a valid controller name, while creating an empty cgroup
+        # on cgroup v2 is unified setup. But it doesn't have a mount point,
+        # lets use the first available controller mount point to form the path
+        if ctrl_name is None:
+            if ignore_systemd:
+                raise IndexError('Unsupported controller \'None\'')
+
+            controllers = ['cpu', 'memory', 'io', 'pids', 'cpuset']
```

Comment:
Does it make sense to have a list of controllers here, or does it make more sense to make `get_controller_mount_point()` smarter and be able to handle a `None` controller?

Reply:
@drakenclimber smart :) Let me implement it

Reply:
Something like below:
```
--- a/tests/ftests/cgroup.py
+++ b/tests/ftests/cgroup.py
@@ -741,6 +741,10 @@ class Cgroup(object):
             for line in mntf.readlines():
                 mnt_path = line.split()[1]
 
+                if ctrl_name is None:
+                    if line.split()[0] == 'cgroup2':
+                        return mnt_path
+
                 if line.split()[0] == 'cgroup':
                     for option in line.split()[3].split(','):
                         if option == ctrl_name:
@@ -758,6 +762,9 @@ class Cgroup(object):
                 mnt_path = line.split()[1]
 
                 if line.split()[0] == 'cgroup2':
+                    if ctrl_name is None:
+                        return mnt_path
+
                     ctrl_file = os.path.join(line.split()[1],
                                              'cgroup.controllers')
 
@@ -774,7 +781,16 @@ class Cgroup(object):
 
     @staticmethod
     def get_controller_mount_point(ctrl_name):
-        vers = CgroupVersion.get_version(ctrl_name)
+        if ctrl_name is None:
+            mode = int(Cgroup.get_cgroup_mode(None))
+            # map return modes:
+            # CGROUP_MODE_LEGACY/CGROUP_MODE_HYBRID ->  CgroupVersion.CGROUP_V1
+            # CGROUP_MODE_UNIFIED -> CgroupVersion.CGROUP_V2
+            if mode >= 2:
+                mode = mode - 1
+            vers = CgroupVersion(mode)
+        else:
+            vers = CgroupVersion.get_version(ctrl_name)
 
         if vers == CgroupVersion.CGROUP_V1:
             return Cgroup.__get_controller_mount_point_v1(ctrl_name)
```